src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy

```diff
@@ -597,6 +597,66 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         flow && flowHelperV2.deleteFlow(flow.flowId)
     }
 
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Able to update a flow endpoint for a single-swithc flow using partial update"() {
+        given: "An active single-switch flow"
+        def swPair = topologyHelper.singleSwitchPair
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+
+        when: "Update the flow: port number and vlan id on the src endpoint"
+        def flowInfoFromDb = database.getFlow(flow.flowId)
+        def ingressCookie = flowInfoFromDb.forwardPath.cookie.value
+        def egressCookie = flowInfoFromDb.reversePath.cookie.value
+        def newPortNumber = topology.getAllowedPortsForSwitch(topology.activeSwitches.find {
+            it.dpId == flow.source.switchId
+        }).last()
+        def newVlanId = flow.source.vlanId + 1
+        flowHelperV2.partialUpdate(flow.flowId, new FlowPatchV2().tap {
+            source = new FlowPatchEndpoint().tap {
+                portNumber = newPortNumber
+                vlanId = newVlanId
+            }
+        })
+
+        then: "Flow is really updated"
+        with(northboundV2.getFlow(flow.flowId)) {
+            it.source.portNumber == newPortNumber
+            it.source.vlanId == newVlanId
+        }
+
+        and: "Flow is valid and pingable"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+        with(northbound.pingFlow(flow.flowId, new PingInput())) {
+            it.forward.pingSuccess
+            it.reverse.pingSuccess
+        }
+
+        and: "The switch passes switch validation"
+        with(northbound.validateSwitch(flow.source.switchId)) { validation ->
+            validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+        }
+        def switchIsFine = true
+
+        and: "The ingress/egress rules are really updated"
+        Wrappers.wait(RULES_INSTALLATION_TIME + WAIT_OFFSET) {
+            def swRules = northbound.getSwitchRules(flow.source.switchId).flowEntries
+            with(swRules.find { it.cookie == ingressCookie }) {
+                it.match.inPort == newPortNumber.toString()
+            }
+            with(swRules.find { it.cookie == egressCookie }) {
+                it.instructions.applyActions.flowOutput == newPortNumber.toString()
+                it.instructions.applyActions.setFieldActions*.fieldValue.contains(newVlanId.toString())
+            }
+        }
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        !switchIsFine && northbound.synchronizeSwitch(flow.source.switchId, true)
```

Comment:
It will be a good idea to have a similar test for `singleSwitchSinglePortFlow`, since it is another special case different from this one. Maybe a 'where' iteration here? If not possible a separate test will be fine

Reply:
added in the `where` block